odoo/models.py

```diff
@@ -5379,22 +5380,32 @@ def convert_pgerror_unique(model, fields, info, e):
     if not ufields or model._table != table:
         return {'message': tools.ustr(e)}
 
+    del cr  # otherwise _() would get the closed cursor from above
     # TODO: add stuff from e.diag.message_hint? provides details about the constraint & duplication values but may be localized...
     if len(ufields) == 1:
         field_name = ufields[0]
-        field = fields[field_name]
-        message = _(u"The value for the field '%s' already exists (this is probably '%s' in the current model).") % (field_name, field['string'])
+        field_description = _convert_field_name(model, field_name)
+        message = _(u"The value for the field '%s' already exists (this is probably '%s' in the current model).") % (field_name, field_description)
         return {
             'message': message,
             'field': field_name,
         }
-    field_strings = [fields[fname]['string'] for fname in ufields]
+    field_strings = [_convert_field_name(model, fname) for fname in ufields]
     message = _(u"The values for the fields '%s' already exist (they are probably '%s' in the current model).") % (', '.join(ufields), ', '.join(field_strings))
     return {
         'message': message,
         # no field, unclear which one we should pick and they could be in any order
     }
 
+def _convert_field_name(model, field_name):
+    """From a lang-free environment, build a new one with the information from
+       the current user, to get the field(_name)'s description in its language.
+    """
+    field = model._fields[field_name]
+    env = model.with_context(model.env.user.context_get()).env
```

Comment:
This should probably merge the "current context" back in, so that if the current context *does* have a lang we're correctly using it and not overriding with the user's default.